internal/plugin/service/externalidentity/external_identity_data_source.go

```diff
@@ -0,0 +1,141 @@
+package externalidentity
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aiven/aiven-go-client/v2"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/aiven/terraform-provider-aiven/internal/plugin/util"
+)
+
+var (
+	_ datasource.DataSource              = &externalIdentityDataSource{}
+	_ datasource.DataSourceWithConfigure = &externalIdentityDataSource{}
+
+	_ util.TypeNameable = &externalIdentityDataSource{}
+)
```

Comment:
Some formatting issue in `TypeNameable` line

```suggestion
var (
	_ datasource.DataSource              = &externalIdentityDataSource{}
	_ datasource.DataSourceWithConfigure = &externalIdentityDataSource{}
	_ util.TypeNameable                  = &externalIdentityDataSource{}
)
)
```